xc/common/utils/prjxray_create_ioplace.py

```diff
@@ -96,46 +96,46 @@ def main():
     # to be used in fasm2bels.
     iostandard_constraints = {}
 
-    fname = args.pcf.name.replace(".pcf", ".json")
-    if os.path.isfile(fname):
-        with open(fname, "r") as fp:
-            iostandard_constraints = json.load(fp)
-
+    if args.pcf:
+        fname = args.pcf.name.replace(".pcf", ".json")
+        if os.path.isfile(fname):
+            with open(fname, "r") as fp:
+                iostandard_constraints = json.load(fp)
+    if args.pcf:
+        pcf_constraints = parse_simple_pcf(args.pcf)
+        net_to_pad = [(constr.net, constr.pad) for constr in pcf_constraints]
+    else:
+        net_to_pad = io_place.net_to_pad
```

Comment:
Should this code merge `io_place.net_to_pad` and `net_to_pad = [(constr.net, constr.pad) for constr in pcf_constraints]`, and generate an error if there is a conflict?

Reply:
@litghost You mean to enable adding the constraints from pcf and xdc simultaneously? Wasn't the plan to drop PCFs once we migrate to XDC? Hence the if...else

Reply:
Your code doesn't enforce either or at the argument management level.  The current code will silently ignore one or the other, depending on the order of the if's.  That strategy isn't acceptable.  Merging the incoming constraints is the most flexible strategy, but if we want to have exclusive arguments, you have to enforce that as an error.

Reply:
Ok, implemented the merge.